### PR TITLE
Move units and validation to separate artifacts

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: units-validation/src/main/java/io/airlift/unitsvalidation/MaxDataSize.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.airlift.units;
+package io.airlift.unitsvalidation;
```
Comment (findepi) [inline units-validation/src/main/java/io/airlift/unitsvalidation/MaxDataSize.java]: It needs to not be on the list:
https://github.com/prestosql/presto/blob/8be3d79e34c1ea5df1a05ff64904e358f2be9a04/presto-main/src/main/java/io/prestosql/server/PluginManager.java#L65-L71

File: units/pom.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    <parent>
+        <groupId>io.airlift</groupId>
+        <artifactId>units-root</artifactId>
+        <version>1.4-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>units</artifactId>
+    <name>units</name>
+    <description>Backward compatible package which pulls every part of units as dependency</description>
```
Comment (findepi) [inline units/pom.xml]: This didn't work out when testing locally. 
Maybe just drop this artifact?

**Comment by findepi** [comment]
This change isn't backward compatible (cannot be).
And validation cannot be under `io.airlift.units.` package (that's the whole point of the change actually). I picked `io.airlift.unitsvalidation`, but I'm not happy with it.

I propose
- put units in `io.airlift.units.units`
- put validation in `io.airlift.units.validation`
- bump version to 2.0

cc @electrum

**Comment by findepi** [comment]
Alternatively validation can go into `io.airlift.validation` (and optionally move to airlift/airlift)

**Comment by electrum** [comment]
The Presto plugin classloader could be changed to do an exact match against the package name rather than suffix.

**Comment by findepi** [comment]
Correct. But wouldn't that be brittle?

**Comment by electrum** [comment]
The engine needs to understand exactly what is part of the plugin classloader. That list needs to be kept in sync with the actual dependencies whenever they change. It's not ideal, but seems like a better solution than an ugly package name and backwards incompatible change for everyone.

**Pull request closed without merging**